recipe/meta.yaml

```diff
@@ -86,17 +89,23 @@ outputs:
         - setuptools_scm
         - wheel
         - numpy
-        - pyarrow
+        # https://github.com/single-cell-data/TileDB-SOMA/issues/1926
+        - pyarrow <13.0 # [osx]
```

Comment:
I suspect that the error loading S3 objects described in https://github.com/single-cell-data/TileDB-SOMA/issues/1926#issuecomment-1834695149 is a runtime-only error. However, given the complexity of the [libarrow run_exports](https://github.com/conda-forge/arrow-cpp-feedstock/blob/62c529770482209a33e3efd93c16a8779eda181f/recipe/meta.yaml#L162), I agree it is safest to also enforce these pins at build time

Reply:
@jdblischak my thoughts exactly